packages/ra-core/src/controller/list/useList.spec.tsx

```diff
@@ -315,4 +315,47 @@ describe('<useList />', () => {
             );
         });
     });
+
+    describe('onSelectAll', () => {
+        it('should select all items if no items are selected', async () => {
+            const children = jest.fn();
+            const data = [{ id: 0 }, { id: 1 }];
+            render(<UseList data={data} callback={children} />);
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelectAll();
```

Comment:
You're using an implementation detail to trigger the action. Instead, add a button to your story so that you can use a simple target in your test. It will also allow to do some visual check.